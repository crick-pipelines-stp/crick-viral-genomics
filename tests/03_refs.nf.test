nextflow_pipeline {
    name "Test Pipeline Components"
    script "../main.nf"

    tag "refs"

    test("refs_host_genome_resolve_illumina") {
        tag "refs/host_genome_resolve_illumina"
        when {
            params {
                run_remove_host_reads = true
                run_reporting         = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp_min.csv"
                host_genome = "sacCer3"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("refs_host_genome_resolve_nanopore") {
        tag "refs/host_genome_resolve_nanopore"
        when {
            params {
                run_remove_host_reads = true
                run_minimap_align     = true
                run_reporting         = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp_min.csv"
                host_genome = "sacCer3"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("refs_merge_refs") {
        tag "refs/merge_refs"
        when {
            params {
                run_reporting = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("refs_force_upper") {
        tag "refs/force_upper"
        when {
            params {
                run_reporting      = false
                force_ref_to_upper = true

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

        test("refs_annotate_flu_ref") {
        tag "refs/annotate_flu_ref"
        when {
            params {
                run_reporting    = false
                annotate_flu_ref = true

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/H1N1_NA_Uk/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp_single.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }
}
