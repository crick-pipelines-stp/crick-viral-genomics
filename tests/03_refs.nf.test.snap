{
    "refs_snapgene_convert": {
        "content": [
            3,
            [
                "00_reference",
                "00_reference/SARS-CoV-2.fasta.fai",
                "00_reference/gff.gff",
                "pipeline_info"
            ],
            [
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2",
                "gff.gff:md5,c795e81dd05cd89d40fd9c7aa526a387"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-08T16:22:55.623314"
    },
    "refs_host_genome_resolve_nanopore": {
        "content": [
            46,
            [
                "00_reference",
                "00_reference/SARS-CoV-2.fasta.fai",
                "01_qc",
                "01_qc/mosdepth",
                "01_qc/mosdepth/FAY66992_BC15.mosdepth.global.dist.txt",
                "01_qc/mosdepth/FAY66992_BC15.mosdepth.region.dist.txt",
                "01_qc/mosdepth/FAY66992_BC15.mosdepth.summary.txt",
                "01_qc/mosdepth/FAY66992_BC15.per-base.bed.gz",
                "01_qc/mosdepth/FAY66992_BC15.per-base.bed.gz.csi",
                "01_qc/mosdepth/FAY66992_BC15.regions.bed.gz",
                "01_qc/mosdepth/FAY66992_BC15.regions.bed.gz.csi",
                "01_qc/mosdepth/FAY66992_BC23.mosdepth.global.dist.txt",
                "01_qc/mosdepth/FAY66992_BC23.mosdepth.region.dist.txt",
                "01_qc/mosdepth/FAY66992_BC23.mosdepth.summary.txt",
                "01_qc/mosdepth/FAY66992_BC23.per-base.bed.gz",
                "01_qc/mosdepth/FAY66992_BC23.per-base.bed.gz.csi",
                "01_qc/mosdepth/FAY66992_BC23.regions.bed.gz",
                "01_qc/mosdepth/FAY66992_BC23.regions.bed.gz.csi",
                "02_contaminant_removal",
                "02_contaminant_removal/FAY66992_BC15.host.sorted.bam",
                "02_contaminant_removal/FAY66992_BC15.host.sorted.bam.bai",
                "02_contaminant_removal/FAY66992_BC23.host.sorted.bam",
                "02_contaminant_removal/FAY66992_BC23.host.sorted.bam.bai",
                "02_contaminant_removal/host_stats",
                "02_contaminant_removal/host_stats/FAY66992_BC15.host.flagstat",
                "02_contaminant_removal/host_stats/FAY66992_BC15.host.idxstats",
                "02_contaminant_removal/host_stats/FAY66992_BC15.host.stats",
                "02_contaminant_removal/host_stats/FAY66992_BC23.host.flagstat",
                "02_contaminant_removal/host_stats/FAY66992_BC23.host.idxstats",
                "02_contaminant_removal/host_stats/FAY66992_BC23.host.stats",
                "03_viral_reads",
                "03_viral_reads/01_host_removed",
                "03_viral_reads/01_host_removed/FAY66992_BC15.viral_1.fastq.gz",
                "03_viral_reads/01_host_removed/FAY66992_BC15.viral_other.fastq.gz",
                "03_viral_reads/01_host_removed/FAY66992_BC23.viral_1.fastq.gz",
                "03_viral_reads/01_host_removed/FAY66992_BC23.viral_other.fastq.gz",
                "06_alignment",
                "06_alignment/FAY66992_BC15.viral.sorted.bam",
                "06_alignment/FAY66992_BC15.viral.sorted.bam.bai",
                "06_alignment/FAY66992_BC23.viral.sorted.bam",
                "06_alignment/FAY66992_BC23.viral.sorted.bam.bai",
                "06_alignment/seqkit",
                "06_alignment/seqkit/FAY66992_BC15.bam_info.tsv",
                "06_alignment/seqkit/FAY66992_BC23.bam_info.tsv",
                "06_alignment/stats",
                "06_alignment/stats/FAY66992_BC15.viral.flagstat",
                "06_alignment/stats/FAY66992_BC15.viral.idxstats",
                "06_alignment/stats/FAY66992_BC15.viral.stats",
                "06_alignment/stats/FAY66992_BC23.viral.flagstat",
                "06_alignment/stats/FAY66992_BC23.viral.idxstats",
                "06_alignment/stats/FAY66992_BC23.viral.stats",
                "06_alignment/unfiltered",
                "06_alignment/unfiltered/FAY66992_BC15.bam",
                "06_alignment/unfiltered/FAY66992_BC23.bam",
                "08_consensus",
                "08_consensus/count_tables",
                "08_consensus/count_tables/FAY66992_BC15.count_table.csv",
                "08_consensus/count_tables/FAY66992_BC23.count_table.csv",
                "08_consensus/pileup",
                "08_consensus/pileup/FAY66992_BC15.mpileup",
                "08_consensus/pileup/FAY66992_BC23.mpileup",
                "pipeline_info"
            ],
            [
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2",
                "FAY66992_BC15.mosdepth.global.dist.txt:md5,6195674f07536bae082a2294ba9a1e0a",
                "FAY66992_BC15.mosdepth.region.dist.txt:md5,d308017cd4b34ada62791d7ee797f3d5",
                "FAY66992_BC15.mosdepth.summary.txt:md5,17f9bb05000ba60e46fb0e98bca1a909",
                "FAY66992_BC15.per-base.bed.gz:md5,f3135daedb4421f1b53d2cbd50e17bf7",
                "FAY66992_BC15.per-base.bed.gz.csi:md5,d5bb67b949b5e4468f6e5b369a9be248",
                "FAY66992_BC15.regions.bed.gz:md5,0fc16c16b3496433a9fbf970e9e2ec1d",
                "FAY66992_BC15.regions.bed.gz.csi:md5,79d96fadc01130495010da50f8adc4bf",
                "FAY66992_BC23.mosdepth.global.dist.txt:md5,6188eb472e87df6d79647e277d778f5a",
                "FAY66992_BC23.mosdepth.region.dist.txt:md5,055e0d824f541ebf873620f8d534b667",
                "FAY66992_BC23.mosdepth.summary.txt:md5,d96df29f2ac4f8740cae74ec8444d1da",
                "FAY66992_BC23.per-base.bed.gz:md5,8cae3a32bae11411abe236e47e563941",
                "FAY66992_BC23.per-base.bed.gz.csi:md5,6a077791382405e1800c84771bc0df0f",
                "FAY66992_BC23.regions.bed.gz:md5,3be751f1633be09285ce3a0ca7e0fb52",
                "FAY66992_BC23.regions.bed.gz.csi:md5,a9ec10413d8726d5ad0efb689a6b7d1c",
                "FAY66992_BC15.host.sorted.bam:md5,5f1f5acf7fd084155ce7b6cb6db16289",
                "FAY66992_BC15.host.sorted.bam.bai:md5,e91dce50260434b47f311e6867b12bcf",
                "FAY66992_BC23.host.sorted.bam:md5,ca37e7346fa99f73524282552a362327",
                "FAY66992_BC23.host.sorted.bam.bai:md5,b500b9e5bbbfa3d1f00391c1d2049d82",
                "FAY66992_BC15.host.flagstat:md5,e08521543a095801f48f59c6884aa135",
                "FAY66992_BC15.host.idxstats:md5,646c153849e6691daa9e1a81e13dddc0",
                "FAY66992_BC15.host.stats:md5,9f477847d976dd870d67d362a4dfa02e",
                "FAY66992_BC23.host.flagstat:md5,e2060f460793696320d456702d2b3ef4",
                "FAY66992_BC23.host.idxstats:md5,c66dbaa2850b0c0d889befe292b5db20",
                "FAY66992_BC23.host.stats:md5,7a9f72a103f004d551de931b0afb6032",
                "FAY66992_BC15.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC15.viral_other.fastq.gz:md5,3b9bf260b4742d7042f23613076d8d2e",
                "FAY66992_BC23.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC23.viral_other.fastq.gz:md5,0d2b1d25c435ec17256f556c112c55b4",
                "FAY66992_BC15.viral.sorted.bam:md5,00d8c211d65e71a7001b92b8a8e4eff1",
                "FAY66992_BC15.viral.sorted.bam.bai:md5,3a3307a8f770c39225d41fa8aee463c8",
                "FAY66992_BC23.viral.sorted.bam:md5,8d9904bf9bd6086b5fd1be92aec4695d",
                "FAY66992_BC23.viral.sorted.bam.bai:md5,afca81de5556a106c71f8413cd019907",
                "FAY66992_BC15.bam_info.tsv:md5,281e857b28f3dac07fbb19a338a120a4",
                "FAY66992_BC23.bam_info.tsv:md5,bc9c90c91481ee0685160355f75705dc",
                "FAY66992_BC15.viral.flagstat:md5,8bf98acbe4b2e82f31b7dc8a2e96bf8c",
                "FAY66992_BC15.viral.idxstats:md5,651859d1dcb738716c3c3988d129a1c2",
                "FAY66992_BC15.viral.stats:md5,8f5408754f58444aa35bda2989463cd8",
                "FAY66992_BC23.viral.flagstat:md5,bbb2ea905eb7c224b82b26e683f5ab5b",
                "FAY66992_BC23.viral.idxstats:md5,101c4e6628c8c676ec9fbcd3b7068ede",
                "FAY66992_BC23.viral.stats:md5,49a9d7b764b6ad50de27432e643e32f1",
                "FAY66992_BC15.bam:md5,9101c7cf919880deb9d6a22f7d96338c",
                "FAY66992_BC23.bam:md5,d153dcc371495b7a054dad89f6b5d9f1",
                "FAY66992_BC15.count_table.csv:md5,2b30361076514aef1a80f2a8e65129f3",
                "FAY66992_BC23.count_table.csv:md5,72f5cbb26f026a12f9c03caa12a76989",
                "FAY66992_BC15.mpileup:md5,367d27a2038044bd29cf881bf1370d1f",
                "FAY66992_BC23.mpileup:md5,afd87541ba07362b83a97031294396c9"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-05-09T10:54:00.753024"
    },
    "refs_force_upper": {
        "content": [
            4,
            [
                "00_reference",
                "00_reference/viral_reference.merged.fasta",
                "00_reference/viral_reference.upper.fasta",
                "00_reference/viral_reference.upper.fasta.fai",
                "pipeline_info"
            ],
            [
                "viral_reference.merged.fasta:md5,887fe7e90382b03d3e9aabb2b03f2ede",
                "viral_reference.upper.fasta:md5,9a39711538a5807ba46cb868b438280e",
                "viral_reference.upper.fasta.fai:md5,c1207acbd8f4d0ce7d301a735c40b378"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-05-09T10:54:20.298921"
    },
    "refs_annotate_flu_ref": {
        "content": [
            10,
            [
                "00_reference",
                "00_reference/viral_reference.merged.fasta",
                "00_reference/viral_reference.merged.fasta.fai",
                "05_ref_annotation",
                "05_ref_annotation/viral_reference",
                "05_ref_annotation/viral_reference.merged.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir",
                "05_ref_annotation/viral_reference/gfflu-outdir/A_United_Kingdom_1_1933_H1N1_NA.split.blastx.tsv",
                "05_ref_annotation/viral_reference/gfflu-outdir/A_United_Kingdom_1_1933_H1N1_NA.split.faa",
                "05_ref_annotation/viral_reference/gfflu-outdir/A_United_Kingdom_1_1933_H1N1_NA.split.gbk",
                "05_ref_annotation/viral_reference/gfflu-outdir/A_United_Kingdom_1_1933_H1N1_NA.split.miniprot.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir/A_United_Kingdom_1_1933_H1N1_NA.split.snpeff.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_HA.split.blastx.tsv",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_HA.split.faa",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_HA.split.gbk",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_HA.split.miniprot.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_HA.split.snpeff.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_NS.split.blastx.tsv",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_NS.split.faa",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_NS.split.gbk",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_NS.split.miniprot.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_NS.split.snpeff.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_PA.split.blastx.tsv",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_PA.split.faa",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_PA.split.gbk",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_PA.split.miniprot.gff",
                "05_ref_annotation/viral_reference/gfflu-outdir/H1N1pdm_PA.split.snpeff.gff",
                "pipeline_info"
            ],
            [
                "viral_reference.merged.fasta:md5,a9cd47d54a178c320b7313e984cf5461",
                "viral_reference.merged.fasta.fai:md5,4d31c9b8cc8ba62c4d00a12b7a141df3",
                "A_United_Kingdom_1_1933_H1N1_NA.split.blastx.tsv:md5,79fc6d2035d682bbc6dc419b3fa7c689",
                "A_United_Kingdom_1_1933_H1N1_NA.split.faa:md5,876b93dbe0607e38068a41148588c34b",
                "A_United_Kingdom_1_1933_H1N1_NA.split.gbk:md5,796f141d6a0a94dbb09ee2e8b70f4f89",
                "A_United_Kingdom_1_1933_H1N1_NA.split.miniprot.gff:md5,91fc69ffb599589086c686e752cc3472",
                "A_United_Kingdom_1_1933_H1N1_NA.split.snpeff.gff:md5,84dbe2f954f4bcf972e09d84795be366",
                "H1N1pdm_HA.split.blastx.tsv:md5,d214d26634d45c2111bead751565a25c",
                "H1N1pdm_HA.split.faa:md5,6146b2cfe236b9caab32c2074595efb8",
                "H1N1pdm_HA.split.gbk:md5,d450c2ff2d629f3c90b81c72e88433ad",
                "H1N1pdm_HA.split.miniprot.gff:md5,75dc04cbd6f2bf6e7c5e0f82039ceb2b",
                "H1N1pdm_HA.split.snpeff.gff:md5,6cb60aa158eb3aa3da83b8fb078324c3",
                "H1N1pdm_NS.split.blastx.tsv:md5,0fa897089a88e1b60f97508625e5bda9",
                "H1N1pdm_NS.split.faa:md5,55e7c4a97e1b792c442309c4abc42e4f",
                "H1N1pdm_NS.split.gbk:md5,a9fabae1fb3dcc0d86736aded6d8da00",
                "H1N1pdm_NS.split.miniprot.gff:md5,f4c95ee3d21498911e22889fb3a464ae",
                "H1N1pdm_NS.split.snpeff.gff:md5,b2b157ba7b472f11e65426b3e6f231d7",
                "H1N1pdm_PA.split.blastx.tsv:md5,7fef68014e0ad5ea39ef0c20eff9dd44",
                "H1N1pdm_PA.split.faa:md5,9c3a56665a36361cbdd8d61f8a19339d",
                "H1N1pdm_PA.split.gbk:md5,1385a2582056b6a1a1703b776dc0648f",
                "H1N1pdm_PA.split.miniprot.gff:md5,108ac8b6950b4a1fd915f7ec8f36cb47",
                "H1N1pdm_PA.split.snpeff.gff:md5,76e5f5e8ad9d2d2733a2d2bf12652215"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-14T16:07:05.041001"
    },
    "refs_host_genome_resolve_illumina": {
        "content": [
            20,
            [
                "00_reference",
                "00_reference/SARS-CoV-2.fasta.fai",
                "02_contaminant_removal",
                "02_contaminant_removal/FAY66992_BC15.host.sorted.bam",
                "02_contaminant_removal/FAY66992_BC15.host.sorted.bam.bai",
                "02_contaminant_removal/FAY66992_BC23.host.sorted.bam",
                "02_contaminant_removal/FAY66992_BC23.host.sorted.bam.bai",
                "02_contaminant_removal/host_stats",
                "02_contaminant_removal/host_stats/FAY66992_BC15.host.flagstat",
                "02_contaminant_removal/host_stats/FAY66992_BC15.host.idxstats",
                "02_contaminant_removal/host_stats/FAY66992_BC15.host.stats",
                "02_contaminant_removal/host_stats/FAY66992_BC23.host.flagstat",
                "02_contaminant_removal/host_stats/FAY66992_BC23.host.idxstats",
                "02_contaminant_removal/host_stats/FAY66992_BC23.host.stats",
                "03_viral_reads",
                "03_viral_reads/01_host_removed",
                "03_viral_reads/01_host_removed/FAY66992_BC15.viral_1.fastq.gz",
                "03_viral_reads/01_host_removed/FAY66992_BC15.viral_other.fastq.gz",
                "03_viral_reads/01_host_removed/FAY66992_BC23.viral_1.fastq.gz",
                "03_viral_reads/01_host_removed/FAY66992_BC23.viral_other.fastq.gz",
                "pipeline_info"
            ],
            [
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2",
                "FAY66992_BC15.host.sorted.bam:md5,cb5201e40b6c665c2c09277cee3e281c",
                "FAY66992_BC15.host.sorted.bam.bai:md5,8e6acc7f50292c77c31c2434daff7b1c",
                "FAY66992_BC23.host.sorted.bam:md5,7131b8ce8fb08e0cc3c47b5498371e43",
                "FAY66992_BC23.host.sorted.bam.bai:md5,b500b9e5bbbfa3d1f00391c1d2049d82",
                "FAY66992_BC15.host.flagstat:md5,97c4d446d23e33217115d33debfdd7fb",
                "FAY66992_BC15.host.idxstats:md5,e1c6d291c21fd6a4e3ebf2e0b56bdeba",
                "FAY66992_BC15.host.stats:md5,d8672e15b5110c7722bbb854e2067a73",
                "FAY66992_BC23.host.flagstat:md5,e2060f460793696320d456702d2b3ef4",
                "FAY66992_BC23.host.idxstats:md5,c66dbaa2850b0c0d889befe292b5db20",
                "FAY66992_BC23.host.stats:md5,7a9f72a103f004d551de931b0afb6032",
                "FAY66992_BC15.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC15.viral_other.fastq.gz:md5,d56568ed4fcf527d52fcedbd7b530fc4",
                "FAY66992_BC23.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC23.viral_other.fastq.gz:md5,0d2b1d25c435ec17256f556c112c55b4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-05-09T10:53:32.836614"
    },
    "refs_merge_refs": {
        "content": [
            3,
            [
                "00_reference",
                "00_reference/viral_reference.merged.fasta",
                "00_reference/viral_reference.merged.fasta.fai",
                "pipeline_info"
            ],
            [
                "viral_reference.merged.fasta:md5,887fe7e90382b03d3e9aabb2b03f2ede",
                "viral_reference.merged.fasta.fai:md5,c1207acbd8f4d0ce7d301a735c40b378"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-14T16:06:42.219565"
    }
}